src/maingoblint.ml

```diff
@@ -155,18 +156,13 @@ let basic_preprocess ~all_cppflags fname =
   (* The actual filename of the preprocessed sourcefile *)
   let nname =  Filename.concat !Goblintutil.tempDirName (Filename.basename fname) in
   if Sys.file_exists (get_string "tempDir") then
-    nname
+    (nname, None)
```

Comment:
This case seems a bit un-intuitive, when does that happen?

Reply:
Apparently the `tempDir` option is for "Reuse temporary directory for preprocessed files". So it assumes that the preprocessed file already exists in the temporary directory.

I've been thinking about generalizing the concept to also apply to compilation database preprocesing, but I'd leave that for another PR.